src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -542,6 +510,56 @@ pass::EliminateEltwise::EliminateEltwise() {
     this->register_matcher(m, callback);
 }
 
+NGRAPH_RTTI_DEFINITION(pass::EliminateReshape, "EliminateReshape", 0);
+
+pass::EliminateReshape::EliminateReshape() {
+    MATCHER_SCOPE(EliminateReshape);
+    auto reshape_node_pattern = pattern::wrap_type<opset8::Reshape>();
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+        auto node = m.get_match_root();
+        auto input = node->input_value(0);
+        // check if reshape is not identity op
+        if (input.get_partial_shape().is_dynamic() || node->get_output_partial_shape(0).is_dynamic()) {
+            NGRAPH_DEBUG << node << " has dynamic shapes.";
+            return false;
+        }
+        // remove identity op
+        if (input.get_shape() == node->get_output_shape(0)) {
+            return replace_output_update_name(node->output(0), input);
+        }
+        // eliminate redundant reshape, squeeze, or unsqueeze
+        auto input_node = input.get_node_shared_ptr();
+        if (ov::as_type_ptr<opset3::Squeeze>(input_node) ||
+            ov::as_type_ptr<opset3::Unsqueeze>(input_node) ||
+            ov::as_type_ptr<opset3::Reshape>(input_node)) {
+            auto shape = node->get_output_shape(0);
+
+            // remove interchangeable reshape, squeeze, or unsqueeze
+            if (input_node->get_input_partial_shape(0).is_static() && input_node->get_input_shape(0) == shape) {
+                node->output(0).get_node_shared_ptr()->set_friendly_name(node->get_friendly_name());
+                copy_runtime_info({input_node, node}, node->output(0).get_node_shared_ptr());
```

Comment:
Usually we set friendly name and rt_info if we ***insert some layer instead of*** root node. In this case, we don't insert anything. So we don't need to set rt_info and friendly name to the layer after root node

Reply:
Got it, thanks! 53ae67c4aa75f461c2a260eeb96473f8186380c9